addons/project/wizard/project_share_wizard.py

```diff
@@ -11,7 +11,14 @@ class ProjectShareWizard(models.TransientModel):
 
     @api.model
     def default_get(self, fields):
-        result = super().default_get(fields)
+        # We pass active_id and active_model in context because super method gets value in context.
+        ctx = dict(self._context)
+        default_res_model = self._context.get('default_res_model')
+        default_res_id = self._context.get('default_res_id')
+        default_access_mode = self._context.get('default_access_mode')
+        if default_res_model and default_res_id:
+            ctx.update({'active_model': default_res_model, 'active_id': default_res_id, 'default_access_mode': default_access_mode, })
+        result = super(ProjectShareWizard, self.with_context(ctx)).default_get(fields)
```

Comment:
Then, I suggest to do that instead of your changes:
```suggestion
        # The project share action could be called in `project.collaborator`
        # and so we have to check the active_model and active_id to use
        # the right project.
        active_model = self._context.get('active_model', '')
        active_id = self._context.get('active_id', False)
        if active_model == 'project.collaborator':
            active_model = 'project.project'
            active_id = self._context('default_project_id', False)
        result = super(ProjectShareWizard, self.with_context(active_model=active_model, active_id=active_id)).default_get(fields)
```